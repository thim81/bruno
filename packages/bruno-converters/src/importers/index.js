export { importCollection as importPostmanCollection } from './postman-collection.js';
export { importEnvironment as importPostmanEnvironment } from './postman-environment.js';
export { importCollection as importInsomniaCollection } from './insomnia-collection.js';
export { importCollection as importOpenAPICollection } from './openapi-collection.js';

export { exportCollection as exportBrunoCollection } from './collections/export';
