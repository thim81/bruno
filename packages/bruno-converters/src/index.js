export { importCollection as importPostmanCollection } from './importers/postman-collection.js';
export { importEnvironment as importPostmanEnvironment } from './importers/postman-environment.js';
export { importCollection as importInsomniaCollection } from './importers/insomnia-collection.js';
export { importCollection as importOpenAPICollection } from './importers/openapi-collection.js';

export { exportCollection as exportBrunoCollection } from './collections/export';
