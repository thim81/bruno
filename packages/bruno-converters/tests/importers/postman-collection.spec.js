import { describe, it, expect } from '@jest/globals';
import path from 'path';
import { importPostmanCollection } from '../../src';

describe('postman-collection', () => {
  it('should correctly import a valid Postman collection file', async () => {
    // Path to the sample Postman collection file
    const fileName = path.resolve(__dirname, '../data', 'sample_postman_collection.json');

    // Call the importCollection function with the sample file
    const brunoCollection = await importPostmanCollection(fileName);

    // Assert that the returned collection matches the expected structure
    expect(brunoCollection).toMatchSnapshot()
  });
});
