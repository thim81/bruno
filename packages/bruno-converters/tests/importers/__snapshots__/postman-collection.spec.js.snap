// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postman-collection should correctly import a valid Postman collection file 1`] = `
{
  "environments": [],
  "items": [
    {
      "items": [
        {
          "name": "Sample GET Request",
          "request": {
            "auth": {
              "awsv4": null,
              "basic": null,
              "bearer": null,
              "mode": "none",
            },
            "body": {
              "formUrlEncoded": [],
              "json": null,
              "mode": "none",
              "multipartForm": [],
              "text": null,
              "xml": null,
            },
            "docs": undefined,
            "headers": [],
            "method": "GET",
            "params": [
              {
                "description": undefined,
                "enabled": true,
                "name": "foo1",
                "type": "query",
                "uid": "mockeduuidvalue123456",
                "value": "bar1",
              },
              {
                "description": undefined,
                "enabled": true,
                "name": "foo2",
                "type": "query",
                "uid": "mockeduuidvalue123456",
                "value": "bar2",
              },
            ],
            "url": "https://postman-echo.com/get?foo1=bar1&foo2=bar2",
          },
          "seq": 1,
          "type": "http-request",
          "uid": "mockeduuidvalue123456",
        },
        {
          "name": "Sample POST Request",
          "request": {
            "auth": {
              "awsv4": null,
              "basic": null,
              "bearer": null,
              "mode": "none",
            },
            "body": {
              "formUrlEncoded": [],
              "json": "{ "key": "value" }",
              "mode": "json",
              "multipartForm": [],
              "text": null,
              "xml": null,
            },
            "docs": undefined,
            "headers": [
              {
                "description": undefined,
                "enabled": true,
                "name": "Content-Type",
                "uid": "mockeduuidvalue123456",
                "value": "application/json",
              },
            ],
            "method": "POST",
            "params": [],
            "url": "https://postman-echo.com/post",
          },
          "seq": 2,
          "type": "http-request",
          "uid": "mockeduuidvalue123456",
        },
      ],
      "name": "Sample Folder",
      "type": "folder",
      "uid": "mockeduuidvalue123456",
    },
    {
      "name": "Standalone Request",
      "request": {
        "auth": {
          "awsv4": null,
          "basic": null,
          "bearer": null,
          "mode": "none",
        },
        "body": {
          "formUrlEncoded": [],
          "json": null,
          "mode": "none",
          "multipartForm": [],
          "text": null,
          "xml": null,
        },
        "docs": undefined,
        "headers": [],
        "method": "GET",
        "params": [],
        "url": "https://postman-echo.com/status/200",
      },
      "seq": 1,
      "type": "http-request",
      "uid": "mockeduuidvalue123456",
    },
  ],
  "name": "Sample Postman Collection with Folder",
  "uid": "mockeduuidvalue123456",
  "version": "1",
}
`;
